Display current and total cell info to the user

keverets opened the pull request:
I like to see that even when I'm not getting WiFi APs, there's still valuable info being stumbled.  This is a first stab at satisfying issue #194 ... though that wanted individual cell tower, this is currently counting unique "Cell Info" which includes things like Signal Strength.

The equality and hashCode for CellInfo could be changed to exclude non-tower-identifying info if required.

keverets committed:
Add getCellInfoCount() and getCurrentCellInfoCount() to CellScanner

keverets committed:
Add "Current Cell Info" and "Cell Info Scanned" to main activity

keverets committed:
Override equals() and hashCode() for CellInfo

Djfe commented:
can you draw the celltowers on the map (test mozilla location service button)?
I know atleast one other app (like mozstumbler) that draws them (by recognizing where they are) and even adds specific data like names/ids of the cell towers

keverets committed:
Reword "Cell Info Scanned" and "Current Cell Info" to "Cells Scanned" and "Cells"

"Cell Info" would be confusing to users, and though they're not strictly
speaking "Cells" but rather readings from cells, this seemed a reasonable
compromise.

cpeterso commented:
Thanks, @keverets!

cpeterso commented:
@keverets: After testing your cell info count, I was wonder why the stumbler claimed that it had scanned of thousands of cells in one day. I now realize what you meant above by "cell info" vs "cell" because the counter is displaying the number of different _scans_, not unique cells. I think we might want to rename the display title or, better yet in my opinion, only display the number of unique cells scanned, like we do for Wi-Fi APs.

Djfe commented:
so that is the reason why my cell count climbs up so fast and my visible cell count switched between 1 and another number all the time :)
I was really wondering about that for some time

so the cell towers just get scanned in a certain Intervall and during scanning only the used tower is really visible

I'm agreeing with you that it would be better to only display scanned unique cells

btw. if you check 2g only in the options of your Android device
will it be impossible to detect 3g cells for mozstumbler?

while we are at it: do you think it would make sense to differentiate between scanned 2g/3g/4g cells, and to differentiate between scanned b/g/n/ac wifi aps in the list

keverets commented:
@cpeterso Sorry for not being more clear about the issue.  I'll look at making it track actual Cells rather than Cell Readings (which include the signal strength, etc).  Would you prefer a new "Cell" object to be created that could be incorporated into the CellInfo, or keep the objects as they are and change the equivalence/hashing of CellInfo to disregard non-identifying Cell information (again, such as Signal Strength)?

cpeterso commented:
@keverets: I think a new `Cell` object that stores the cell's identifying information sounds like a good idea. So a `CellInfo` would have-a `Cell` and measurements like signal strength and scan timestamp? CellScanner.java only uses `mCells` to track the number of scanned cells, so `mCells` would only need to store `Cell`s, not `CellInfos`.

Djfe commented:
Still not fixed I think.
The cells scanned counter still doesn't show only unique cell towers.
